[GlobalISel] Use replaceRegOrBuildCopy when legalizer-combining anyext(undef).

davemgreen opened the pull request:
This just avoids the unnecessary creation of some COPY nodes created from the CSE builder.

arsenm commented:
Make clang-format happy by using buildUndef instead of the low level buildInstr?

davemgreen committed:
[GlobalISel] Use replaceRegOrBuildCopy when legalizer-combining anyext(undef).

This avoids the unnecessary creation of some COPY nodes created from the CSE
builder.
